kotlinx-coroutines-core/jvm/test/linearizability/LockFreeListLCStressTest.kt

```diff
@@ -44,29 +44,11 @@ class LockFreeListLCStressTest : TestBase() {
     private fun Any.isSame(value: Int) = this is Node && this.value == value
 
     @Test
-    fun testAddRemoveLinearizability() {
-        val options = StressOptions()
-            .iterations(100 * stressTestMultiplierSqrt)
-            .invocationsPerIteration(1000 * stressTestMultiplierSqrt)
-            .threads(3)
-        LinChecker.check(LockFreeListLCStressTest::class.java, options)
-    }
-
-    private var _curElements: ArrayList<Int>? = null
-    private val curElements: ArrayList<Int> get() {
-        if (_curElements == null) {
-            _curElements = ArrayList()
-            q.forEach<Node> { _curElements!!.add(it.value) }
-        }
-        return _curElements!!
-    }
-
-    override fun equals(other: Any?): Boolean {
-        other as LockFreeListLCStressTest
-        return curElements == other.curElements
-    }
+    fun testAddRemoveLinearizability() = LCStressOptionsDefault().check(this::class)
 
-    override fun hashCode(): Int {
-        return curElements.hashCode()
+    override fun extractState(): Any {
+        val elements = ArrayList<Int>()
+        q.forEach<Node> { elements.add(it.value) }
+        return elements
```

Comment:
It would be also very useful to add call to `q.validate()` here which also checks that the resulting doubly-linked list has consistent links in both directions.

Reply:
This code is used only for verification, to specify a state machine. Unfortunately, _lincheck_ does not have a mechanism to validate the data structure at the end of each invocation; we can add this feature to _licheck_.

Reply:
We don't have to wait for new lincheck. We can add it later.